discord_connector.py

```diff
@@ -0,0 +1,566 @@
+
+
+    def fetch_message(self, channel_id, message_id, action_result) -> discord.Message or None:
+        status, channel = self.run_in_loop(self._guild.fetch_channel(channel_id), action_result,
+                                           message="Cannot fetch channel from Discord.")
+        if not status:
+            return status, None
+        status, message = self.run_in_loop(channel.fetch_message(message_id), action_result,
+                                           message="Cannot fetch message from Discord.")
+
+        return status, message
+
+
+    def create_artifacts(self, message):
+        container_id = self.get_container_id()
+        attachments = []
+        embeds = []
+
+        if message.embeds:
+            self.save_progress("working on embeds")
+            for embed in message.embeds:
+                embeds.append(self.create_embed_artifact(embed, container_id))
+
+        if message.attachments:
+            self.save_progress("working on attachments")
+            for attachment in message.attachments:
+                attachments.append(self.create_attachment_artifact(attachment, container_id))
+
+        return attachments, embeds
+
+
+    def create_embed_artifact(self, embed, container_id):
+        artifact = Artifact(
+            container_id=container_id,
+            name=f"embed: {embed.title}",
+            cef={"URL": embed.url, "Description": embed.description}
+        )
+        return self.save_artifact_to_soar(dataclasses.asdict(artifact))
+
+
+    def create_attachment_artifact(self, attachment, container_id):
+        artifact = Artifact(
+            container_id=container_id,
+            name=f"embed: {attachment.title}",
+            cef={"URL": attachment.url, "Description": attachment.description, "Type": attachment.content_type}
+        )
+        return self.save_artifact_to_soar(dataclasses.asdict(artifact))
+
+
+    def save_artifact_to_soar(self, artifact):
+        status, creation_message, artifact_id = self.save_artifact(artifact)
+        self.save_progress("creating artifact: status: {}, creation message: {}, artifact id {}"
+                           .format(status, creation_message, artifact_id))
+        return artifact_id
+
+
+    def parse_message(self, message, attachments, embeds):
+        return {
+            "message origin": {
+                "channel id": message.channel.id,
+                "channel name": message.channel.name,
+            },
+            "message data": {
+                "created at": str(message.created_at),
+                "edited at": str(message.edited_at) if message.edited_at is not None else "message was not edited",
+            },
+            "author data": {
+                "author id": message.author.id,
+                "author name": message.author.name,
+            },
+            "jump url": message.jump_url,
+            "flags": list(filter(lambda flag: getattr(message.flags, flag), MESSAGE_FLAGS)) or "no flags",
```

Comment:
Do we need to have this list - MESSAGE_FLAGS? Only thing here that we don't want to show is `value` property (https://discordpy.readthedocs.io/en/latest/api.html#messageflags)

Also, iterator mentioned in the docs should give you a pair of flags, so you don't need to use kinda hackhy `getattr`

Reply:
It has been changed so now it does not use const and getattr